src/main/java/com/google/sps/servlets/AggregationServlet.java

```diff
@@ -40,30 +45,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.getWriter().println(e.getMessage());
       return;
     }
+
     try {
       List<ChromeOSDevice> devices = amassDevices();
-      MultiKeyMap<String, Integer> data = processData(devices, fields);
+      MultiKeyMap<String, List<String>> data = processData(devices, fields);
       response.setStatus(HttpServletResponse.SC_OK);
       response.getWriter().println(Json.toJson(new AggregationResponse(data, fields)));
-    } catch (IOException e) {
-      response.sendRedirect("/login");
-      return;
-    } catch (Exception e) {
-      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-      response.getWriter().println(e.getMessage());
-      return;
-    }
+    } catch (IOException e) {//something went wrong during getting the devices
+        response.sendRedirect("/authorize.html");
+    } catch (TooManyResultsException e) {
+        response.sendRedirect("/login.html");
```

Comment:
I remember you mentioned it earlier, but thinking about the user experience, they were trying to aggregate by fields but after a few moments, they sent back to login page without any reason, which is not ideal. We should send back response with internal error state and let UI display what went wrong.

Please leave a todo to change it later.

Reply:
Please make the TODO comment with some details. The only word is "Return". It's hard for the collaborators to understand what "Return" means.